Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -15,13 +19,42 @@ public sealed class ZombieSystem : EntitySystem
         [Dependency] private readonly DiseaseSystem _disease = default!;
         [Dependency] private readonly BloodstreamSystem _bloodstream = default!;
         [Dependency] private readonly ZombifyOnDeathSystem _zombify = default!;
+        [Dependency] private readonly ServerInventorySystem _inv = default!;
         [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly ILogManager _logManager = default!;
         public override void Initialize()
         {
             base.Initialize();
+
             SubscribeLocalEvent<ZombieComponent, MeleeHitEvent>(OnMeleeHit);
         }
 
+        private float GetZombieInfectionChance(EntityUid uid, ZombieComponent component)
+        {
+            float baseChance = component.MaxZombieInfectionChance;
+
+            if (!TryComp<InventoryComponent>(uid, out var inventoryComponent))
+                return baseChance;
+
+            if (!_inv.TryGetContainerSlotEnumerator(uid, out var enumerator, inventoryComponent))
+                return baseChance;
+
+            var items = 0f;
+            var total = 0f;
+            while (enumerator.MoveNext(out var con))
+            {
+                total++;
+                if (con.ContainedEntity != null)
+                    items++;
+            }
+
+            var max = component.MaxZombieInfectionChance;
```

Comment:
This is pretty weird because stuff like pocket items will block infection chances.

Reply:
I'm aware that this is kind of weird, but it doesn't really impact much. It can also be considered Depth:tm: